Item 9782: Text Choice data type support for field editor updates to in use values

cnathe opened the pull request:
#### Rationale
For this round of changes to the Text Choice data type epic, we are adding support for allowing updates to in use text choice values from within the field editor UI along with locking those values that are in use by a "locked" item (i.e. sample with a locked status). This PR includes the server side updates to update the rows for any in use text choice values which are updated in the field editor UI. The DomainUtil class has helpers to use the query update server for the given domain table to call the updateRows() method, which will give us the detailed audit logging we want for this action.

#### Related Pull Requests
* https://github.com/LabKey/labkey-ui-components/pull/697
* https://github.com/LabKey/platform/pull/2903
* https://github.com/LabKey/biologics/pull/1109
* https://github.com/LabKey/sampleManagement/pull/789

#### Changes
* update core module @labkey/components version
* during the DomainUtil.updateDomainDescriptor() method, detect text choice validator value updates and call updateRows() to remap values in those rows

cnathe committed:
update @labkey/components package version

cnathe committed:
DomainUtil helper to update text choice values during updateDomain() call

cnathe committed:
update @labkey/components package version

cnathe committed:
update @labkey/components package version

labkey-susanh commented on internal/src/org/labkey/api/exp/property/DomainUtil.java:
```diff
@@ -972,6 +993,48 @@ else if (prop == null)
+            if (domainTable != null && domainTable.getUpdateService() != null)
+            {
+                // we need to make all of the row updates for this domain property at one time to prevent the
+                // double mapping if one choice value want from a -> b and another from b -> c
```
```suggestion
                // double mapping if one choice value went from a -> b and another from b -> c
```

cnathe replied:
done

cnathe committed:
update @labkey/components package version

cnathe committed:
Merge branch 'develop' into fb_textChoiceUpdates

# Conflicts:
#	core/package-lock.json
#	core/package.json

cnathe committed:
update @labkey/components package version

cnathe committed:
update @labkey/components package version

cnathe committed:
Filter out aliquots from query to get rows to update for text choice value change

cnathe committed:
update @labkey/components package version

cnathe committed:
update @labkey/components package version

cnathe committed:
remove extra "_row" prop from audit event on text choice value change

cnathe committed:
Add containerFilter param to domain kind getTableInfo() method

cnathe committed:
Account for shared domains in text choice value updates
- query using all container filter
- make one updateRows call per container in resulting rows

cnathe committed:
Merge branch 'develop' into fb_textChoiceUpdates

cnathe committed:
update @labkey/components package version

cnathe committed:
update @labkey/components package version

cnathe committed:
update @labkey/components package version

cnathe committed:
update @labkey/components package version
